### PR TITLE
Specify a versions of NuGet packages without using a temporary file

### PR DESCRIPTION
**Pull request by Taritsyn**
Currently, the `compiled.nuspec` temporary file is used to dynamically specify a versions of NuGet packages. I suggest using the [built-in feature](https://docs.microsoft.com/en-us/nuget/reference/nuspec#replacement-tokens) of NuGet package manager instead of creating a temporary file.

### CODE REVIEW

**Review by ppenzin — COMMENTED** [review]
This looks good to me, though I have no NuGet experience. Maybe we should set up a deploy job for 'nightly' binaries, this would help us test this.

**Comment by Taritsyn** [comment]
Hello, Petr!

This is a good idea! On the `chakracore-preview` MyGet feed has not published NuGet packages for a long time.

So far, it makes sense to publish only two packages: `Microsoft.ChakraCore` and `Microsoft.ChakraCore.vc140`. `Microsoft.ChakraCore.Symbols` package definitely cannot be published.

**Comment by rhuanjl** [comment]
We're not going to be able to publish as Microsoft.ChakraCore in the future - we'll have to use a different name as Microsoft have given the project to the community it's not going to be them publishing it for any future release (other than maybe 2 or 3 more security patches for 1.11).

**Review by ppenzin — APPROVED** [review]

**Comment by ppenzin** [comment]
Hi Andrey,

Again, I am not savvy with NuGet at all, so if you don't mind we could probably use some help from somebody who understands it. As Richard pointed out, we need to rename the packages. Would it be possible to auto-deploy the release build from CI? I assume it would, but I thought I'd ask.

@rhuanjl, should we merge this? I have filed #6564 for package renaming.

**Comment by Taritsyn** [comment]
Hello, Petr!

> Again, I am not savvy with NuGet at all, so if you don't mind we could probably use some help from somebody who understands it.

Basically, I can advise you on creating packages. I've been building a NuGet packages for about 10 years.  In addition, I have experience in creating [packages containing the ChakraCore library](https://github.com/Taritsyn/JavaScriptEngineSwitcher/wiki/ChakraCore).

> Would it be possible to auto-deploy the release build from CI?

For these purposes, you can use the `nuget.exe` (see the [“Publishing packages > Command line”](https://docs.microsoft.com/en-us/nuget/nuget-org/publish-a-package#command-line) subsection of  documentation).

**Comment by Taritsyn** [comment]
Hello, Richard!

> … - we'll have to use a different name as Microsoft have given the project to the community …

This is even better, because we can start from scratch. For example, a separate package could be released for each platform, i.e. perform a [reorganization as in the ClearScript project](https://github.com/microsoft/ClearScript/releases/tag/7.0.0), only without renaming of the assembly files.

**Comment by rhuanjl** [comment]
> @rhuanjl, should we merge this? I have filed #6564 for package renaming.

As a start the change seems fine, one comment is that with the Microsoft CLA defunct we'll need to ask @Taritsyn to sign our new contributor agreement - which means we perhaps need to finalise that first

**Comment by Taritsyn** [comment]
@ppenzin

> For these purposes, you can use the `nuget.exe` …

I forgot to say that because a .NET Framework is not supported on Linux and OS X, `nuget.exe` needs to be run by using a [Mono](https://www.mono-project.com/):

```
mono nuget.exe …
```

**Comment by ppenzin** [comment]
> As a start the change seems fine, one comment is that with the Microsoft CLA defunct we'll need to ask @Taritsyn to sign our new contributor agreement - which means we perhaps need to finalise that first

@rhuanjl - true, maybe we can merge just part, though I am fine with merging the whole PR. When agreement is in, I can start automating its check.

> Basically, I can advise you on creating packages. I've been building a NuGet packages for about 10 years. In addition, I have experience in creating [packages containing the ChakraCore library](https://github.com/Taritsyn/JavaScriptEngineSwitcher/wiki/ChakraCore).

@Taritsyn - thank you for offering your help, will let you know if I have questions.

> For these purposes, you can use the `nuget.exe` (see the [“Publishing packages > Command line”](https://docs.microsoft.com/en-us/nuget/nuget-org/publish-a-package#command-line) subsection of documentation).

This seems to be straight-forward. I will try after we merge this change in (or maybe earlier, depending on workload).

**Comment by rhuanjl** [comment]
Thanks for signing the contribution agreement @Taritsyn

I merged this without waiting for CI as the CI doesn't test this anyway. Thanks for the contribution.

**Pull request merged**
